analyses/hotspots-detection/utils/filterMaf.R

```diff
@@ -0,0 +1,138 @@
+#' Takes a sqllite table from an individual caller in maf format
+#' and filters as per given user filtering options
+#' @param table a maf format dataframe
+#' @param impact_values list of IMPACT [optional] values to filter
+#' @param hotspot_amino_acid_position_df  [optional] dataframe with Hugo_Sybol, Amino_Acid_Position
+#' and hotspot_database columns
+#' @param hotspot_genomic_site_df [optional] dataframe with Hugo_Symbol and Chromosome, Start_Position ,
+#' End_Position, Hugo_Symbol and hotspot_database columns
+#' @return A dataframe with base columns from maf filtered as per user input for Variant_Classification
+#' column with "variant_classification" OR IMPACT column with "impact_values" OR Hugo_Symbol using "gene_table".
+#' Additional filtering can be done per amino acid position hotspots using "hotspot_amino_acid_site_df"
+#' OR per genomic site hotspots using "hotspot_genomic_site_df"
+#'
+
+filterMaf <- function(table,
+                      impact_values,
+                      hotspot_database_2017_snv_df,
+                      hotspot_database_2017_indel_df,
+                      hotspot_genomic_site_df) {
+  if (!missing(impact_values)) {
+    # IMPACT filtering values
+    table <- table %>%
+      dplyr::filter(IMPACT %in% impact_values)
+  }
+  
+  # filtered calls_base
+  calls_base <- table %>%
+    # only keep annotation for canonical protein_coding transcripts 
+    filter( CANONICAL=='YES' & 
+              BIOTYPE=='protein_coding') %>%
+    as.data.frame() %>%
+    # create an Amino_Acid_Position column
+    dplyr::mutate(
+      Amino_Acid_Position =
+        case_when(
+          # If not splice site we want to capture Amino_Acid_Position to match
+          !grepl("Splice_Site",Variant_Classification) ~
+            # format of Protein_position is 594/766
+            # [Amino_Acid_Position/ Protein_length]
+            # stripping protein length here to get Amino_Acid_Position
+            gsub("/.*", "", Protein_position),
+          # If splice site we want to capture HGVSp_Short to match 
+          grepl("Splice_Site",Variant_Classification) ~
+            # since these sites are intronic sites adding
+            # HGSVp_Short instead of Protein_position
+            gsub("p[.]","",HGVSp_Short))
+    ) %>%
+    mutate(
+      Start_Position = as.numeric(Start_Position),
+      End_Position = as.numeric(End_Position)
+    )
+
+  if (!missing(hotspot_database_2017_snv_df) & !missing(hotspot_database_2017_indel_df)) {
```

Comment:
Just noticing that if one of these is provided but not the other, this will be skipped entirely, which may not be what you want. This logic appears in a few places, so we would want to fix all of them, or potentially fail early if only one of the two is provided.

Reply:
Thanks for the note! I added a snippet which will fail early in the script if only 1 of the 2 hotspots database is provided asking to provide the complete database which is snv(+splice) and indel both.